dev/integration_tests/flutter_driver_screenshot_test/test_driver/driver_screenshot_tester.dart

```diff
@@ -5,7 +5,7 @@
 import 'dart:async';
 import 'dart:io' show File;
 
-import 'package:flutter/foundation.dart';
+import 'package:flutter/src/foundation/collections.dart';
```

Comment:
Is this a terrible idea, @dnfield?  Would it better to copy this helper similar to `_deepEqualsList`?

Reply:
Hm. @goderbauer or @jonahwilliams is this expected?

I don't think we want to do an implementation import here.  What got changed that caused this?

Reply:
I see now.  

I'm not sure whether it's better to copy the implementation or use an impelemntation import here.  We should generally avoid both.

Reply:
You're saying you wouldn't expect foundation to implicitly require the dart:ui package?
The `dart:ui` imports/exports in the foundation package are years old.
```
foundation/basic_types.dart:10:1: Error: Not found: 'dart:ui'
export 'dart:ui' show 

foundation/binding.dart:9:8: Error: Not found: 'dart:ui'
import 'dart:ui' as ui show saveCompilationTrace, Window, window;

foundation/key.dart:5:8: Error: Not found: 'dart:ui'
import 'dart:ui' show hashValues;\

foundation/binding.dart:87:3: Error: Type 'ui.Window' not found.
ui.Window get window => ui.window;

foundation/change_notifier.dart:68:20: Error: Type 'VoidCallback' not found.
void addListener(VoidCallback listener);

foundation/change_notifier.dart:72:23: Error: Type 'VoidCallback' not found.
void removeListener(VoidCallback listener);
```
see https://github.com/flutter/flutter/issues/47700 for full list.

Reply:
We talked offline.  Driver shouldn't depend on foundation, agreed just writing function to do comparison is better.